src/Workspaces/Core/Portable/NamingStyles/EditorConfig/EditorConfigNamingStyleParser.cs

```diff
@@ -59,10 +60,28 @@ public static NamingStylePreferences ParseDictionary(IReadOnlyDictionary<string,
                 }
             }
 
-            return new NamingStylePreferences(
+            var preferences = new NamingStylePreferences(
                 symbolSpecifications.ToImmutableAndFree(),
                 namingStyles.ToImmutableAndFree(),
                 namingRules.ToImmutableAndFree());
+
+            var orderedRules = preferences.Rules.NamingRules
```

Comment:
Comment the intent please. Sticking your email contents here would be *fantastic* since it motivated a why behind these sets of rules.